opentracing-api/src/main/java/io/opentracing/Tracer.java

```diff
@@ -20,113 +20,154 @@
+     */
+    <C> SpanContext extract(Format<C> format, C carrier);
+
+    interface SpanBuilder extends SpanContext {
+
+        /**
+         * A shorthand for addReference(References.CHILD_OF, parent).
+         */
+        SpanBuilder asChildOf(SpanContext parent);
+
+        /**
+         * A shorthand for addReference(References.CHILD_OF, parent.context()).
+         */
+        SpanBuilder asChildOf(Span parent);
+
+        /**
+         * Add a reference from the Span being built to a distinct (usually parent) Span. May be called multiple times to
+         * represent multiple such References.
+         * <p>
+         * If no references are added manually before {@link SpanBuilder#start()} is invoked, an
+         * {@link References#INFERRED_CHILD_OF} reference is created to any {@link Scheduler#activeContext()}
+         * context.
+         *
+         * @param referenceType the reference type, typically one of the constants defined in References
+         * @param referencedContext the SpanContext being referenced; e.g., for a References.CHILD_OF referenceType, the
+         *                          referencedContext is the parent
+         *
+         * @see io.opentracing.References
+         */
+        SpanBuilder addReference(String referenceType, SpanContext referencedContext);
+
+        /** Same as {@link Span#setTag(String, String)}, but for the span being built. */
+        SpanBuilder withTag(String key, String value);
+
+        /** Same as {@link Span#setTag(String, boolean)}, but for the span being built. */
+        SpanBuilder withTag(String key, boolean value);
+
+        /** Same as {@link Span#setTag(String, Number)}, but for the span being built. */
+        SpanBuilder withTag(String key, Number value);
+
+        /** Specify a timestamp of when the Span was started, represented in microseconds since epoch. */
+        SpanBuilder withStartTimestamp(long microseconds);
+
+        /**
+         * Returns a newly started and {@linkshort Scheduler.Continuation#activate(boolean) activated}
+         * {@link Scheduler.Continuation}.
+         *
+         * <p>
+         *
+         * Note that the Continuation supports try-with-resources. For example:
+         * <pre>{@code
+           try (Scheduler.Continuation spanCont = tracer.buildSpan("...").startAndActivate(true)) {
+               // Do work
+               Span span = tracer.scheduler().active();
+               span.setTag( ... );  // etc, etc
+           }
+           }</pre>
+         *
+         * @param finishOnDeactivate if true, the {@link Span} encapsulated by the {@link Scheduler.Continuation} will
+         *                   finish() upon invocation of {@link Scheduler.Continuation#deactivate()}.
+         * @return a pre-activated {@link Scheduler.Continuation}
+         *
+         * @see Scheduler.Continuation#activate(boolean)
+         */
+        Scheduler.Continuation startAndActivate(boolean finishOnDeactivate);
```

Comment:
This is new, and maybe should take precedence over plain-old `start()` for most use cases.